src/cpp/src/model_runner.hpp

```diff
@@ -92,33 +111,35 @@ class ModelRunner {
                     position_ids_data[token_id] = position_id;
                 }
 
-                past_lens_data[0] = group_context_len;
+                size_t expected_kv_cache_size = sequence_group->get_num_processed_tokens() - sequence->get_num_evicted_tokens();
```

Comment:
when we have performed eviction, should not we just decrease num_processed (read as "context len") tokens? In this case, we don't need to subtract a number of evicted tokens here

Reply:
On the one hand, this is loss of information that could be used further, on the other - less chance for overflow. I propose to keep the subtraction unless it impacts performance drastically